protocol/directory.go

```diff
@@ -110,6 +110,12 @@ func (d *ConiksDirectory) Register(req *RegistrationRequest) (
 		d.tbs[req.Username] = tb
 		return NewRegistrationProof(ap, d.LatestSTR(), tb, Success)
 	} else {
+		currentAP, _ := d.pad.LookupInCurrentEpoch(req.Username)
+		if bytes.Equal(currentAP.LookupIndex, currentAP.Leaf.Index) {
+			// use the ap from the latest STR to return to the client
+			return NewRegistrationProof(ap, d.LatestSTR(), nil, ErrorNameExisted)
```

Comment:
It's probably fine to continue to implement TBs as an extension (since we can still imagine other solutions) but, for the moment, we should panic if `useTBs` is `false`.